### ISSUE TITLE
Where do I place additional dependencies (AtomicParsley.exe, phantomjs.exe)? Where do I place the patched ffmpeg binaries?

### ISSUE THREAD

**Issue by fmotion1** [yt-dlp/yt-dlp#7731]
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I *intentionally* remove or skip any mandatory\* field

### Checklist

- [X] I'm asking a question and **not** reporting a bug or requesting a feature
- [X] I've looked through the [README](https://github.com/yt-dlp/yt-dlp#readme)
- [X] I've verified that I'm running yt-dlp version **2023.07.06** ([update instructions](https://github.com/yt-dlp/yt-dlp#update)) or later (specify commit)
- [X] I've searched [known issues](https://github.com/yt-dlp/yt-dlp/issues/3766) and the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar questions **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)

### Please make sure the question is worded well enough to be understood

I am on: Windows 10 Pro x64 Version 22H2 (OS Build 19045.3208)
I have the following Python versions installed:

```
 -V:3.11 *        C:\Python\Python311\python.exe
 -V:3.10          C:\Python\Python310\python.exe
 -V:3.6           C:\Python\Python36\python.exe
 -V:2.7           C:\Python\Python27\python.exe
 -V:2.7-32        C:\Python\Python27x32\python.exe
 ```

Strap in, because there are a lot of questions here! 🙃

1. I downloaded the latest [yt-dlp.exe](https://github.com/yt-dlp/yt-dlp/releases/latest/download/yt-dlp.exe).
2. I placed that exe into a folder that has been added to my system `PATH` environment variable.
3. I noticed that two additional dependencies are not included with this release. These dependencies are: [AtomicParsley](https://github.com/wez/atomicparsley), and [phantomjs](https://github.com/ariya/phantomjs).
4. I downloaded these dependencies (binary exe files). 
5. How do I make sure these binaries are integrated / detected / or otherwise leveraged by `yt-dlp`?
6. Is this even possible, since I downloaded the pre-compiled binary?

I also have some questions about the **Python** version.

1. I created a `yt-dlp` VENV with version `3.11 x64` of Python, opened a shell, and activated the VENV.
2. I ran the command `pip install -U yt-dlp` (I don't need the `python3 -m` part, as I'm inside an active VENV.) 
3. The installation went smoothly and everything is up to date in my VENV.
4. Now, as before, how do I add the two additional dependencies? ([AtomicParsley](https://github.com/wez/atomicparsley), and [phantomjs](https://github.com/ariya/phantomjs)? These dependencies are only offered as exe files (as far as I can tell), so am I correct in thinking that they need to be moved to a specific place for `yt-dlp` to "detect" them?
5. Where do I place the `yt-dlp` ffmpeg binaries so they work?
6. Furthermore, what is the difference between the Python install and the binary (exe) install? Should I use the python version of `yt-dlp` instead of the binary?
7. Can I compile the Python version to a single non-shared executable after adding the two additional assemblies?

Lastly, a couple questions about the **custom ffmpeg** builds. (`yt-dlp/FFmpeg-Builds`)

1. I downloaded [ffmpeg-master-latest-win64-gpl.zip](https://github.com/yt-dlp/FFmpeg-Builds/releases/download/latest/ffmpeg-master-latest-win64-gpl.zip) from `yt-dlp/FFmpeg-Builds`
2. Where do I place these binaries? How does `yt-dlp` detect and leverage them? Is it enough to drop them into my `BIN` directory? (My BIN directory is added to my system `PATH` environment variable)
3. Lastly, if I am adding these custom builds to `PATH`, how can I ensure that both the official release of ffmpeg doesn't conflict with the `yt-dlp` specific ffmpeg release? If I add both to path there will be problems. 

---

### Basically, TL;DL 
Where do I place the patched `ffmpeg` binaries? Where do I place `AtomicParsley.exe` and `phantomjs.exe`?

### Provide verbose output that clearly demonstrates the problem

- [ ] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [ ] If using API, add `'verbose': True` to `YoutubeDL` params instead
- [ ] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
This content is not applicable to my question as I'm not having problems running `yt-dlp`.
```

**Comment by bashonly** [comment]
> 5. How do I make sure these binaries are integrated / detected / or otherwise leveraged by `yt-dlp`?
> 6. Is this even possible, since I downloaded the pre-compiled binary?

Yes, put them in here:
> **a folder that has been added to my system `PATH` environment variable.**

---

> 4. Now, as before, how do I add the two additional dependencies? ([AtomicParsley](https://github.com/wez/atomicparsley), and [phantomjs](https://github.com/ariya/phantomjs)? These dependencies are only offered as exe files (as far as I can tell), so am I correct in thinking that they need to be moved to a specific place for `yt-dlp` to "detect" them?
> 5. Where do I place the `yt-dlp` ffmpeg binaries so they work?

same answer as above, in your `PATH`

---

> 6. Furthermore, what is the difference between the Python install and the binary (exe) install? Should I use the python version of `yt-dlp` instead of the binary?

exe has a built-in python interpreter and py dependencies included. Python install will instead run in your python environment

---

> 7. Can I compile the Python version to a single non-shared executable after adding the two additional assemblies?

you mean atomicparsley and phantomjs? why?

---

> 2. Where do I place these binaries? How does `yt-dlp` detect and leverage them? Is it enough to drop them into my `BIN` directory? (My BIN directory is added to my system `PATH` environment variable)

yes, in your `PATH` or in same folder as yt-dlp.exe

---

> 3. Lastly, if I am adding these custom builds to `PATH`, how can I ensure that both the official release of ffmpeg doesn't conflict with the `yt-dlp` specific ffmpeg release? If I add both to path there will be problems. 

if they are in separate directories that are both in your `PATH`, one will have priority over the other. If you only want to use the yt-dlp ffmpeg build with yt-dlp (and not for any other purpose), you can keep it out of your `PATH` and instead pass the `--ffmpeg-location` option to yt-dlp (with yt-dlp ffmpeg.exe's path as the argument). you could create a [config file](https://github.com/yt-dlp/yt-dlp#configuration) and add this option it to make it easier

**Comment by pukkandan** [comment]
> 7\. Can I compile the Python version to a single non-shared executable after adding the two additional assemblies?

Technically yes. See pyinstaller docs for how to add arbitrary files to compilation. But this would result in a large and slow executable. So you probably shouldn't

---

You can check `yt-dlp -v` to check if the exe files are being detected by yt-dlp correctly. It'll look something like:

> [debug] exe versions: ffmpeg n6.0-26-g3f345ebf21-20230708 (setts), ffprobe n6.0-26-g3f345ebf21-20230708, phantomjs 2.1.1

btw, AtomicParsley and phantomjs are not needed by most people. They are only used in some rare situations and if they are needed, yt-dlp will tell you with an appropriate error/warning.

**Comment by jxu** [comment]
Also phantomjs has been discontinued for several years. Whatever used it should be updated.

**Comment by pukkandan** [comment]
https://github.com/yt-dlp/yt-dlp/issues/1354
